### PR TITLE
Document CS8892 warning

### PR DESCRIPTION
**Pull request by Youssef1313**
~~**Don't merge before https://github.com/dotnet/roslyn/pull/46832 gets merged first**.~~ (It's merged now 🎉 )

### CODE REVIEW

File: docs/csharp/language-reference/compiler-messages/cs8892.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
+---
+title: "Compiler warning (level 5) CS8892"
+ms.date: 07/20/2015
+f1_keywords: 
+  - "CS8892"
+helpviewer_keywords: 
+  - "CS8892"
+author: Youssef1313
+---
```
Comment (Youssef1313) [inline docs/csharp/language-reference/compiler-messages/cs8892.md:9]: I see other compile-messages have ms.assetid in the metadata. Is it required? What is it for? How to get the correct value for it?

File: docs/csharp/language-reference/compiler-messages/toc.yml [hunk 2]
```diff
@@ -1,4 +1,4 @@
-- name: C# Compiler Errors
+- name: C# compiler messages
```
Comment (Youssef1313) [inline docs/csharp/language-reference/compiler-messages/toc.yml:1]: This toc node contains both errors and warnings. I updated it to C# compiler messages.

File: docs/csharp/language-reference/compiler-messages/toc.yml [hunk 3]
```diff
@@ -273,3 +273,5 @@
     href: cs1591.md
   - name: Compiler Warning (level 4) CS1610
     href: cs1610.md
+  - name: Compiler warning (level 5) CS8892
```
Comment (Youssef1313) [inline docs/csharp/language-reference/compiler-messages/toc.yml:276]: I started to use sentence-casing as it's now the recommended styleguide. Is it okay to be inconsistent with the other toc nodes?

**Comment by Youssef1313** [comment]
@BillWagner, Roslyn's PR got merged. This is ready for review.

File: docs/csharp/language-reference/compiler-messages/cs8892.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/language-reference/compiler-messages/cs8892.md:9]: That's a historical artifact from the migration from MSDN. It's no longer needed.

File: docs/csharp/language-reference/compiler-messages/toc.yml (hunk 3 above)
Comment (BillWagner) [reply docs/csharp/language-reference/compiler-messages/toc.yml:276]: Thanks for asking this.

I think it looks jarring to mix them. What do you think of updating this section of the TOC to match the current style guide?

**Review by BillWagner — COMMENTED** [review]
Commenting so this stays in my queue.

I did review all the changes @Youssef1313 and this is ready to merge once we resolve the TOC question. I'm OK merging this, and starting a new PR for updating that section of the TOC per the style guide, or adding that change to this PR. Let me know which you prefer.

**Review by BillWagner — APPROVED** [review]
This LGTM @Youssef1313 

Thanks again.  I'll :shipit: now.

**Pull request merged**